### DISCUSSION TITLE
version 6.0.5 broke reset call from useForm

### DISCUSSION THREAD

**Discussion by eduleite** [react-hook-form/react-hook-form#2228] [Q&A]
Bill,

I upgraded my project to 6.0.5 and a form with a structure similar to the one from the codesandbox below stopped functioning.

https://codesandbox.io/s/usewatch-bug-605-pm6uw?file=/src/App.tsx

Steps to reproduce:
1) useForm is called with a initial defaultValues property
2) after a little while, some data is fetched from the server and the component asks the form to reset to the data fetched from the server (this is simulated with a setTimeout on the codesandbox)
3) whenever a value is updated in the form, the result from useWatch will always return the defaultValues that were set in step 2, instead of the updated values (inspect the console log to see the values being logged)

The codesandbox above has a working example of the bug. If I change the version back to 6.0.4, it works fine. The problem is with the reset call in the object returned from useForm. After the reset call, useWatch stops working correctly. In my project I refactored the code to not call the reset and it now works, but I think other people might go through the same problem.

**Comment by bluebill1049** [comment]
hey eduleite, thanks for the issue report. is there any UI visual change from switch .4 to .5? looks the same to me, nothing is checked on those raidos.

**Reply by eduleite** [reply]
There is no change. What changes from .4 to .5 is that in .5 the result from useWatch will always return the new defaultValues set with reset.
In .4 useWatch returns correctly.
In order to .5 to work correctly, reset must not be called and defaultValues should have the desired initial state.

**Reply by eduleite** [reply]
Do you want me to open the issue? I'm dumb, should have openned it earlier, because it is clearly a bug.

**Reply by bluebill1049** [reply]
yes open an issue, i will investigate further, probably include more detail as above too.
